File: server/src/services/weatherservice.js
import { credentials } from '@grpc/grpc-js';
import { WeatherService } from './proto/proto.js';
import { promisify } from 'util';

export default async function (city){
    var weatherService = new WeatherService('localhost:9090', credentials.createInsecure());

    weatherService.GetWeather = promisify(weatherService.GetWeather.bind(weatherService));

    const response = await weatherService.GetWeather({city: city});

    console.log("WeatherService Response: "+JSON.stringify(response));

    return response.weather;
}

File: planner/src/main/java/com/cloudapps/practica4/clients/WeatherService.java
package com.cloudapps.practica4.clients;

import java.util.concurrent.Future;

import org.springframework.scheduling.annotation.Async;
import org.springframework.scheduling.annotation.AsyncResult;
import org.springframework.stereotype.Component;

import net.devh.boot.grpc.client.inject.GrpcClient;
import weatherservice.GetWeatherRequest;
import weatherservice.Weather;
import weatherservice.WeatherServiceGrpc.WeatherServiceBlockingStub;

@Component
public class WeatherService {

	@GrpcClient("weatherServer")
	private WeatherServiceBlockingStub client;
	

	@Async
	public Future<Weather> getWeather(String city) throws Exception {
		
		GetWeatherRequest request = GetWeatherRequest.newBuilder()
	            .setCity("Madrid")
	            .build();
	        
		Weather response = client.getWeather(request);

	    System.out.println("Response received from server:\n" + response);
	    return new AsyncResult<Weather>(response);
		
	}	
}
